avro-fastserde/src/codegen/java/com/linkedin/avro/fastserde/generated/deserialization/AVRO_1_4/Array_of_UNION_GenericDeserializer_585074122056792963_585074122056792963.java

```diff
@@ -67,16 +70,19 @@ public IndexedRecord deserializerecord0(Object reuse, Decoder decoder)
             record = new org.apache.avro.generic.GenericData.Record(arrayElemOptionSchema0);
         }
         int unionIndex1 = (decoder.readIndex());
-        if (unionIndex1 == 0) {
-            decoder.readNull();
-        } else {
-            if (unionIndex1 == 1) {
+        switch (unionIndex1) {
+            case  0 :
+                decoder.readNull();
+                break;
+            case  1 :
                 if (record.get(0) instanceof Utf8) {
```

Comment:
i dont like this instanceOf Utf8 check, for several reasons:

1. the record could (in theory) be using Strings, or any other Charsequence
2. what this code really wants to do is a null check

Reply:
Actually, it is important to check specifically for `Utf8` here because it leads to passing the instance into the `Decoder#readString(Utf8 old)` function. If the old instance is not `Utf8` then we pass null, since other types of `CharSequence` are not mutable and can therefore not be reused.

Reply:
oh. can code gen add javadocs? :-) i missed the whole reuse part of this

Reply:
Good question, I don't know :D ...